vip-helpers/class-wpcomvip-restrictions.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Automattic\VIP\Utils;
+
+class WPComVIP_Restrictions {
+	protected static $instance = null;
+
+	public static function instance(): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		add_filter( 'wp_insert_post_data', [ $this, 'restrict_post_author' ], PHP_INT_MAX );
+		add_filter( 'wp_insert_attachment_data', [ $this, 'restrict_post_author' ], PHP_INT_MAX );
+		add_filter( 'wp_dropdown_users_args', [ $this, 'restrict_post_author_dropdown' ], PHP_INT_MAX );
+		add_filter( 'rest_user_query', [ $this, 'restrict_post_author_dropdown' ], PHP_INT_MAX );
+		add_filter( 'coauthors_edit_ignored_authors', [ $this, 'restrict_post_author_dropdown_coauthors' ], PHP_INT_MAX );
+	}
+
+	public function restrict_post_author( array $data ): array {
+		$wpcomvip = get_user_by( 'login', 'wpcomvip' );
+
+		if ( false !== $wpcomvip && isset( $data['post_author'] ) && $data['post_author'] == $wpcomvip->ID ) {
+			$data['post_author'] = get_current_user_id();
```

Comment:
We probably want to check that current user is not `wpcomvip` either.

Reply:
✅